elasticsearch/src/main/java/org/apache/calcite/adapter/elasticsearch/ElasticsearchTable.java

```diff
@@ -309,6 +322,36 @@ String scriptedFieldPrefix() {
     return Linq4j.asEnumerable(hits.hits()).select(getter);
   }
 
+  private String getFieldTypeFromName(String key, List<Map.Entry<String, String>> aggregations)
+      throws JsonProcessingException {
+    ElasticsearchMapping.Datatype type = transport.mapping.mapping().get(key);
+    if (null != type) {
+      return type.name();
+    }
+
+    Optional<Map.Entry<String, String>> entryOptional = aggregations.stream()
+        .filter(agg -> agg.getKey().equals(key))
+        .findAny();
+    if (!entryOptional.isPresent()) {
```

Comment:
what's the point of using Optional, if `null` is required anyways?  
maybe just use find